libs/statistics/MonteCarloPermutationTest.h

```diff
@@ -158,7 +160,19 @@ namespace mkc_timeseries
 	mBaseLineTestStat = _BackTestResultPolicy<Decimal>::getPermutationTestStatistic(mBackTester);
       //std::cout << "Baseline test stat. for original  strategy equals: " <<  mBaseLineTestStat << ", baseline # trades:" << this->getNumClosedTrades (mBackTester) <<  std::endl << std::endl;
 
-      return _ComputationPolicy::runPermutationTest (mBackTester, mNumPermutations, mBaseLineTestStat);
+      // Create instance of computation policy for observer support
+      _ComputationPolicy computationPolicy;
+      
+      // Chain attached observers to the computation policy (pass-through Subject design)
+      std::shared_lock<std::shared_mutex> observerLock(this->m_observersMutex);
+      for (auto* observer : this->m_observers) {
+          if (observer) {
+              computationPolicy.attach(observer);
+          }
+      }
+      observerLock.unlock();
```

Comment:
nit: we can replace this with a scope, this way:
```suggestion
      {
        std::shared_lock<std::shared_mutex> observerLock(this->m_observersMutex);
        for (auto* observer : this->m_observers) {
            if (observer) {
                computationPolicy.attach(observer);
            }
        }
      }
```

Reply:
Made change in latest udpate.